code/modules/vr/vr_control.dm

```diff
@@ -0,0 +1,74 @@
+/obj/machinery/computer/vr_control
+	name = "VR console"
+	desc = "A list and control panel for all virtual servers."
+	icon_screen = "comm_logs"
+	var/making_room = 0
+	light_color = LIGHT_COLOR_DARKGREEN
+
+/obj/machinery/computer/vr_control/Destroy()
+	return ..()
+
+/obj/machinery/computer/vr_control/attack_ai(mob/user)
+	interact(user)
+
+/obj/machinery/computer/vr_control/attack_hand(mob/user)
+	interact(user)
+
+/obj/machinery/computer/vr_control/interact(mob/user)
+	if(user.incapacitated())
+		return 0
+	ui_interact(user)
+
+/obj/machinery/computer/vr_control/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1)
+	ui = nanomanager.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "vr_control.tmpl", "User Created Rooms", 500, 400)
+		ui.open()
+
+/obj/machinery/computer/vr_control/Topic(href, href_list)
+	if(..())
+		return 0
```

Comment:
Trailing return shouldn't be necessary. Proc will just default return `null`

Reply:
I may be wrong but I think that return 0 is needed for the nanoui. It would be nice to get a second opinion on this.

Reply:
Resolved via PM. I thought the proc terminated here for some reason 😖